core/services/ardupilot_manager/ArduPilotManager.py

```diff
@@ -258,3 +292,53 @@ def remove_endpoints(self, endpoints_to_remove: Set[Endpoint]) -> None:
                 raise
 
         self._update_endpoints()
+
+    def get_available_firmwares(self, vehicle: Vehicle) -> List[str]:
+        try:
+            return self.firmware_download.get_available_versions(vehicle, self.current_platform)
+        except Exception as error:
+            logger.exception(f"Unable to retrieve available firmwares: {error}\n")
+            raise
+
+    def install_custom_firmware(self, firmware_path: pathlib.Path) -> None:
+        try:
+            self.mavlink_manager.stop()
+            self.prune_ardupilot_processes()
+            self.firmware_installer.install_firmware(firmware_path)
+            logger.info(f"Succefully installed firmware for {self.current_platform}.")
+        except Exception as error:
+            logger.exception(f"Unable to install firmware: {error}\n")
+            raise
+        finally:
+            self.restart()
+
+    def install_custom_firmware_version(self, vehicle: Vehicle, version: str) -> None:
```

Comment:
The function name does not help compared to `install_custom_firmware`, the first one install one based in a file, this one downloads the file from the web.
Maybe `install_firmware` for this one makes more sense since it's not a custom firmware.

Reply:
Agree. They got somewhat confusing.
Changed to `update_firmware` (latest), `install_firmware` (vehicle + version) and `install_custom_firmware` (custom file).

Reply:
> plus It looks like they could be refactored.
> some thing like:
> 
> ```
> install_firmware_version(version):
>     file_path = fetch_file(...)
>     install_custom_firmware(file_path)
> ```

Those two can be refactored, for sure.
Done.